File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    private static final Set<String> OPERATORS = Set.of("+", "-", "*", "/");

    public boolean isOperator(String s) {
        return OPERATORS.contains(s);
    }

    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String token : tokens) {
            if (isOperator(token)) {
                int right = stack.pop();
                int left = stack.pop();
                switch (token) {
                    case "+" -> stack.push(left + right);
                    case "-" -> stack.push(left - right);
                    case "*" -> stack.push(left * right);
                    case "/" -> stack.push(left / right);
                }
            } else {
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int operate(const string &s,int &a,int &b){
        if(s=="+") return a+b;
        else if(s=="-") return b-a;
        else if(s=="*") return a*b;
        else if(s=="/") return b/a;
        else return -1;
    }
    int evalRPN(vector<string>& tokens) {
        stack<int>st;
        for(string &s:tokens){
            if(s=="+" || s=="-" || s=="*" || s=="/"){
                int operand1=st.top(); st.pop();
                int operand2=st.top(); st.pop();
                int ans=operate(s,operand1,operand2);
                st.push(ans);
            } else{
                st.push(stoi(s));
            }
        }
        return st.top();
    }
};